### PR TITLE
Filter by date range backend

### PR DESCRIPTION
**Pull request by TeerapatTrepopsakulsin**
### Description
- Implement filter by day of week backend #163

### Functional Acceptance Criteria
- `GET activities/?start_date=<date>&end_date=<date>` Return list of activities that take time that in between start_time and end_time (_Both **Inclusive**_). More details about the format in additional information.

### Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation, if applicable.
- [x] My code follows the Coding Guidelines.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please specify)

### Additional Information
date format: `r'(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2})$'`, for example, 2024-01-20.
Example: 
`GET activities/?start_date=2024-10-10` Will return a list of activities from 10 Oct 2024 on.
`GET activities/?start_date=2024-10-15&end_date=2024-10-20` Will return a list of activities from 15 Oct 2024 until 20 Oct 2024 (Inclusive).

### CODE REVIEW

**Comment by Thanawas-Sirilertsathit** [comment]
I already give you a comment. @TeerapatTrepopsakulsin

**Comment by Thanawas-Sirilertsathit** [comment]
Please clarify

**Review by OmegaOoh — APPROVED** [review]
Looks fine

File: activities/tests/test_index.py [hunk 1]
```diff
@@ -137,3 +148,101 @@ def test_search_by_day_of_week(self):
...
+        response = self.client.get(urls.reverse("activities:index") + f"?start_date={act7_day}&end_date={act1_day}")
+        self.assertJSONEqual(response.content, [])
+
+        response = self.client.get(urls.reverse("activities:index") + f"?start_date={act3_day}&end_date=invalid")
+        self.assertJSONEqual(response.content, [json_act3, json_act7])
+
+        response = self.client.get(urls.reverse("activities:index") + f"?start_date=invalid&end_date={act3_day}")
+        self.assertJSONEqual(response.content, [json_act1, json_act2, json_act3])
+
+        response = self.client.get(urls.reverse("activities:index") + "?start_date=9999-99-99&end_date=invalid")
```
[the remark this answers is not in the document]
Comment (Thanawas-Sirilertsathit) [reply activities/tests/test_index.py:229]: I got answer already ok

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Your answer for me is reasonable. I approved.

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Approve everything is fine after you merged

**Pull request merged**